### PR TITLE
Final Swan service updates

### PR DESCRIPTION
**Pull request by kdmukai**
* Production Swan service api credentials
* Remove most debugging statements in the api calls
* Bugfix on `User._services` left nav
* Promote Swan service to "prod" status
* UI display formatting of autowithdrawal setting
* Possible bugfix for Liquid address details
* Moving `ServiceManager` Instantiation out of Specter.__init__()
* Loading for external extensions in DevelopmentConfig only
* build-script adjustments for clarity and right order
* monkey patch pyinstaller rthook in order to workaround https://github.com/pyinstaller/pyinstaller/issues/6537

### CODE REVIEW

File: src/cryptoadvance/specter/services/swan/client.py [hunk 1]
```diff
@@ -284,13 +266,10 @@ def update_autowithdrawal_addresses(
                         "clientId": "specter-dev"
                     },
                     "specter_wallet_alias": "seedsigner_demo"
-                },
```
Comment (k9ert) [inline src/cryptoadvance/specter/services/swan/client.py:287]: `"btcAddresses": []` seems to be disappearing all over. I don't understand why?

File: src/cryptoadvance/specter/services/swan/config.py [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
+    SWAN_CLIENT_ID = "specter-dev"
+    SWAN_CLIENT_SECRET = (
+        "BcetcVcmueWf5P3UPJnHhCBMQ49p38fhzYwM7t3DJGzsXSjm89dDR5URE46SY69j"
+    )
     SWAN_API_URL = "https://dev-api.swanbitcoin.com"
 
 
 class ProductionConfig(BaseConfig):
+    SWAN_CLIENT_ID = "specter"
+    SWAN_CLIENT_SECRET = (
```
Comment (k9ert) [inline src/cryptoadvance/specter/services/swan/config.py:11]: We should make a remark here for semi-educated security-researchers. Like:
```suggestion
    # Conceptionally, we're using PKCE which should not make any password necessary. However, technically, it
    # was too much challenge to avoid the password and so we technically this password is used but it's not 
    # considered to be a real secret. There is no reasonable attack-vector for this secret being public.
    # Obviously, you're happy to prove wrong but please don't complain about hard-coded secrets or so.
    SWAN_CLIENT_SECRET = (
```

File: src/cryptoadvance/specter/user.py [hunk 3]
```diff
@@ -308,8 +308,8 @@ def save_info(self, delete=False):
 
     def add_service(self, service_id: str, autosave: bool = True):
         """Add a Service to the User. Only updates what is listed in the sidebar."""
-        if service_id not in self.services:
-            self.services.append(service_id)
+        if service_id not in self._services:
+            self._services.append(service_id)
```
Comment (k9ert) [inline src/cryptoadvance/specter/user.py:312]: Why is this necessary? Uh, oh, now i see the issue. This is necessary because of my bad implementation of the property:
```
    @property
    def services(self):
        if self._services:
            return self._services
        return []
```
It should be changed to:
```
    @property
    def services(self):
        if not self._services:
            self._services = []    
        return self._services
```
Do you agree?

File: src/cryptoadvance/specter/services/swan/client.py (hunk 1 above)
Comment (kdmukai) [reply src/cryptoadvance/specter/services/swan/client.py:287]: Pablo changed the return data at some point so we no longer get the list of addresses associated with a Swan wallet. We don't need it anyway, as we can always just overwrite the current with a new autowithdrawal call.

File: src/cryptoadvance/specter/user.py (hunk 3 above)
Comment (kdmukai) [reply src/cryptoadvance/specter/user.py:312]: Either way the `User._services` seems a little inconsistent in that it has a "member"-like level of protection in the class but that isolation isn't really accomplishing anything; external code can directly manipulate the `user_obj.services` prop just as easily as if the var was declared as `User.services` without the prop.

But for the `add_service` method itself, it's within `User` which should be free to modify its own internal `self._services` "private" member whenever/wherever it needs to.

I think this all boils down to philosophy/semantics. Bottom line for me is that the left nav is showing the active services once again!

File: src/cryptoadvance/specter/services/swan/config.py (hunk 2 above)
Comment (kdmukai) [reply src/cryptoadvance/specter/services/swan/config.py:17]: Good call. Comment added at the top of the file.

File: src/cryptoadvance/specter/user.py (hunk 3 above)
Comment (kdmukai) [reply src/cryptoadvance/specter/user.py:312]: Applied your patch to the `@property`.

Comment (k9ert) [reply src/cryptoadvance/specter/user.py:312]: Yes, it's not java. The reason i converted it to a property was avoiding a "has not attribute _services" issue. However, not implemented well. It doesn't matter to me that much whether to use internal or external (even though if it 's possible to use external, it's better in order to avoid the above issue) but the main thing was that this revealed my bad naive implementation.
Good that it's fixed!
